proxylib/testparsers/blockparser.go

```diff
@@ -65,22 +66,24 @@ func getBlock(data [][]byte) ([]byte, int, int, error) {
 				// indicating the length of the frame AFTER the ':'
 				if lenUint64, err := strconv.ParseUint(block.String(), 10, 64); err != nil {
 					return block.Bytes(), 0, 0, err
+				} else if lenUint64 > math.MaxInt64 {
+					return block.Bytes(), 0, 0, fmt.Errorf("block length overflow")
```

Comment:
Couple of questions here:
* Why compare a uint64 type against an int64 size? Is there code lower down that is limiting this to the positive 63-bit range or signed integer comparisons below?
* Why is this a problem if `ParseUint(..., 64)` is limiting it to 64 bits? I understand that the result is a `uint` which doesn't have a defined size, but presumably the last parameter to `ParseUint()` should bound that within 64 bits.

The more I look at this particular change, the more I wonder if the complaint is less about the 64-bit size up here but more to do with some comparison against a shorter type below :thinking:

Reply:
> Why compare a uint64 type against an int64 size?

We parse a value using `strconv.ParseUint` (which returns a `uint64`) but we store the value in blockLen (which is an `int`), so any value greater that `math.MaxInt64` will overflow the `int` (assuming a 64 bit platform).

> The more I look at this particular change, the more I wonder if the complaint is less about the 64-bit size up here but more to do with some comparison against a shorter type below 🤔

It's possible. We only run CodeQL against `master` and it can be difficult to trace exactly why CodeQL complains, so it's possible that this change won't fix CodeQL's complaints.

Reply:
Ah right. If two lines down we're casting down to `int(...)` then seems like we should be doing a `math.MaxInt()` here instead?

Reply:
Done. Note that `math.MaxInt` isn't available at the moment as our runtime CI VMs are still running Go 1.16, but I've left a FIXME to switch this to `math.MaxInt` in the code for when we get them upgraded.

In the short term, I've used the max `int64` constant, which is not correct if Cilium is compiled for a 32-bit architecture, but switching to `math.MaxInt` later will fix this.